Capitalize proper nouns

jamillambert opened the pull request:
Taproot, Merkle and Huffman are proper nouns and should be capitalized in strings and docs.

These have been capitalized everywhere.

All cases of `taproot`/`taptree`/`tap tree` in docs and strings have been changed to be `Taproot tree` or ```[`TapTree`]``` as appropriate.

All cases of `tree` have been changed to lower cases, except where it is in title case.

Error messages in `Taproot` have been changed to start with a lowercase character, except when it is a proper noun, and not end with a period.

Close: #2913

tcharding commented:
Except for the line length the commit log of fc40cf099cd90ab731addac2bb9b9914514bea60 is excellent, I knew exactly why the patch was needed and what it did - super easy to review.

tcharding reviewed (dismissed):
ACK a93bc72f10e4e303d1aaa18adba2fd2a0edfaf8f

tcharding commented:
Nice PR!

jamillambert commented:
> Except for the line length the commit log of [fc40cf0](https://github.com/rust-bitcoin/rust-bitcoin/commit/fc40cf099cd90ab731addac2bb9b9914514bea60) is excellent, I knew exactly why the patch was needed and what it did - super easy to review.

Thanks, and sorry.  My editor had the setting switched off.  I have now enabled a 50 char limit for the commit subject and 72 chars for the message.

tcharding commented:
Legend.

Kixunil reviewed (dismissed):
ACK a93bc72f10e4e303d1aaa18adba2fd2a0edfaf8f

jamillambert commented:
Added 3 patches to extend this pull request to fix everything in the corresponding issue.  

Changed the commit messages to split long lines.

Changed the PR title from **Capitalize Taproot** to **Capitalize proper nouns**

tcharding commented:
I love this PR, you are flying! I think the last patch should be removed and done as a separate PR. The reasons being twofold
1. It doesn't fit under the PR description
2. Its more controversial where as the first 5 patches are not

tcharding reviewed (dismissed):
ACK 7e45c73f17ca0624dfdbc699ee749eb6e350fe3f

apoelstra commented:
utACK 7e45c73f17ca0624dfdbc699ee749eb6e350fe3f. Thanks! Changes like this make the crate feel much more professional.

But this needs a rebase now, sorry.

jamillambert commented:
Rebased to remove conflict.

tcharding reviewed (approved):
ACK e8a30bfca7983cdfb892dbe044b9517c8d1991a6

apoelstra reviewed (approved):
ACK e8a30bfca7983cdfb892dbe044b9517c8d1991a6
